src/libraries/System.Text.Encoding.CodePages/src/System/Text/BaseCodePageEncoding.netcoreapp.cs

```diff
@@ -66,12 +66,10 @@ private static unsafe void ReadCodePageIndex(Stream stream, Span<byte> codePageI
                         default:    codePageName = new string(&codePageIndex.CodePageName); break;
                     }
 
-                    string? resourceName = EncodingNLS.GetLocalizedEncodingNameResource(codePageIndex.CodePage);
                     string? displayName = null;
-
-                    if (resourceName != null && resourceName.StartsWith("Globalization_cp_", StringComparison.OrdinalIgnoreCase))
```

Comment:
I don't actually understand what this was trying to do. If resource name ends up looking like `Globalization_cp_` it means `UsingResourceKeys() == true`. Calling into GetResourceString with that just produces the same string again. I think I'm fixing it, but please double check in case I just don't understand what this was originally doing.

Reply:
Yeah, this is strange.  The only thing I can think of is that maybe it was trying to handle some kind of delegation, where the resource value for one code page named another, but if that were actually the case this would need to have been a while loop to follow the potential chain.  @tarekgh, what was the goal here?